### PR TITLE
Clear District Council Search Box Field on Deselect

### LINKED ISSUE #1699: Clear District Council search box field on deselect
### Overview
We need to clear the District Council searchbox field when the user de-selects the council on the UI in order to clarify to the user that the search is no longer active

### Action Items
- [ ] when a council is removed from the search filters term, also clear the CouncilList searchbox

### Resources/Instructions

Relevant Components/Methods
- SearchBar.jsx
- CouncilsList.jsx
- CouncilSelector.jsx::handleDelete()

<details><summary>Demo Video of Problem</summary>
<p>

https://github.com/hackforla/311-data/assets/6414668/936e99ad-33de-40d9-8db8-5053f71bc483

</p>
</details>

### PR DESCRIPTION
**Pull request by drakenguyen4000**
Fixes #1699
### What changes did you make?
  - Made changes to Councilslist.jsx and index.jsx under CouncilSelector folder so when the user de-selects the council it will clear the District Council search box field.  
 
### Why did you make the changes (we will use this info to test)?
  - So when the user de-selects the council it will clear the District Council search box field.  
  - [ ] Up to date with `main` branch
  - [ ] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [ ] All PR Status checks are successful
  - [ ] Peer reviewed and approved
Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

**Review by Skydodle — APPROVED** [review]
@drakenguyen4000 

- Pulled down the PR and tested, visually worked as expected and council name no longer remained after delete clicked
- Reviewed the code: the resetChildState toggle effectively triggered useEffect to reset search term to empty string, then revert value back to false to ready for next interaction. Nice work!!

**Review by ryanfchase — CHANGES_REQUESTED** [review]
Hi @drakenguyen4000 and @Skydodle -- I'm actually thinking we should take a different approach. Sorry for the delay, it's taken a long time to wrap my head around the control flow of the map search filters.

While setting a temporary reset flag on a timer solves the problem, I think we can solve this problem in a more react-y way. Here are some ideas:
1. utilize `createRef` / `useRef` and maintain a member variable for the searchbar in `<CouncilSelector/>` clearing the `value` in `CouncilSelector::handleDelete`. You will drill the ref down to the `<SearchBar/>` in `CouncilsList.jsx`
2. we can move `searchTerm` to be a state variable belonging to `CouncilSelector`, rather than `CouncilsList`, similarly allowing this state to be cleared on `CouncilSelector::handleDelete`
3. we could incorporate `searchTerm` as part of a reducer (e.g. `filters.js` or `ui.js`). Though, I think this would be a bit overkill because currently no other component needs to know what `searchTerm` is.

Let me know your thoughts -- or if there are any follow up questions

**Comment by Skydodle** [comment]
> Hi @drakenguyen4000 and @Skydodle -- I'm actually thinking we should take a different approach. Sorry for the delay, it's taken a long time to wrap my head around the control flow of the map search filters.
> 
> While setting a temporary reset flag on a timer solves the problem, I think we can solve this problem in a more react-y way. Here are some ideas:
> 
> 1. utilize `createRef` / `useRef` and maintain a member variable for the searchbar in `<CouncilSelector/>` clearing the `value` in `CouncilSelector::handleDelete`. You will drill the ref down to the `<SearchBar/>` in `CouncilsList.jsx`
> 2. we can move `searchTerm` to be a state variable belonging to `CouncilSelector`, rather than `CouncilsList`, similarly allowing this state to be cleared on `CouncilSelector::handleDelete`
> 3. we could incorporate `searchTerm` as part of a reducer (e.g. `filters.js` or `ui.js`). Though, I think this would be a bit overkill because currently no other component needs to know what `searchTerm` is.
> 
> Let me know your thoughts -- or if there are any follow up questions

@ryanfchase and @drakenguyen4000 
Hey Ryan thanks for the insightful review and suggestions. I took another look at the code and agreed with you on moving away from the current implementation. While it was a creative solution to the problem, I do see that it can be done in a more react-y way without relying on a temporary state reset flag and a timer.

The centralized state management method in option 2 made the most sense to me:
- Uplift the `searchTerm` state from `<CouncilsList />` to `<CouncilSelector />`
- Use `setSearchTerm` to reset `searchTerm` to an empty string in `handleDelete` 
- Pass down `searchTerm` and `setSearchTerm` as props to `<CouncilsList />`
- Add a handler to handle search change for the `<SearchBar />` in `<CouncilsList />`

I went ahead and implemented this option, and it tested well. This method is cleaner and make use of the `searchTerm` state properly without introducing new complexities, and lastly it is the proper "react-y" way which would be easier for code readability.

**Comment by drakenguyen4000** [comment]
@ryanfchase @Skydodle  Thanks for putting this out guys.  My previous solution was more complicated than it should be.  Yes, moving state up to Councilselector and passing down searchTerm and setSearchTerm to CouncilsList makes things much simpler.

File: components/main/Desktop/CouncilSelector/CouncilsList.jsx [hunk 1]
```diff
@@ -44,9 +44,8 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
-function CouncilsList({ items, onClick }) {
+function CouncilsList({ items, onClick, searchTerm, setSearchTerm }) {
```
Comment (Skydodle) [inline components/main/Desktop/CouncilSelector/CouncilsList.jsx:47]: On line 78 below in CouncilsList.propTypes, don't forget to add the propTypes for `searchTerm` and `setSearchTerm`.

```
searchTerm: PropTypes.string.isRequired;
setSearchTerm: PropTypes.func.isRequired;
```

It helps us maintain type safety and clarity about the expected types of props the component relies on, making it easier for anyone to understand the prop requirement at a glance. Most importantly it gets rid of the annoying eslint red warnings ;)

**Review by Skydodle — CHANGES_REQUESTED** [review]
Just need to add the prop types for `searchTerm` and `setSearchTerm` in `< CouncilsList />`

Everything else looks great and it worked as expected when I tested with your PR pull down.

**Comment by drakenguyen4000** [comment]
@Skydodle  Thanks Johnny.  I just made the changes.

**Review by Skydodle — APPROVED** [review]
LGTM! Thank you @drakenguyen4000

**Review by ryanfchase — APPROVED** [review]
approved, thanks Drake!

**Pull request merged**